### PR TITLE
[invoice] speedup pdf generation with details (sql refactoring)

### PR DESCRIPTION
**Pull request by osallou**
when generating pdf with details on invoice, reduce number of queries and speedup generation

### CODE REVIEW

**Review by gaetanHerve — APPROVED** [review]
tested and ok to me. Found a bug while testing, but not linked to this PR. I make an issue

**Comment by osallou** [comment]
maybe we should first understand #708 before merge?

**Comment by gaetanHerve** [comment]
#708 is independant, does it on develop branch too. I'm looking forward to fix it

**Comment by gaetanHerve** [comment]
but, indeed, we could integrate the fix to this PR

**Comment by osallou** [comment]
Ok, the updaye itself was introduced in my latest fix , replacing labels by index j (was failing due to parenthesis in label)

But j counter was itself in an other loop with i counter. So we could have duplicate names (label_j)

File: Modules/invoices/View/Invoiceglobal/editAction.php [hunk 1]
```diff
@@ -108,23 +103,13 @@
...
-                    price = $('#price_' + j).val();
-
-                    data[i]["data"]["count"][j]["quantity"] = quantity;
-                    data[i]["data"]["count"][j]["unitprice"] = price;
-                }
+            for (let i=0; i<items.length; i++) {
+                quantity = $('#quantity_' + i).val();
+                price = $('#price_' + i).val();
+                items[i].data.quantity = quantity;
+                items[i].data.unitprice = price;
```
[the remark this answers is not in the document]
Comment (gaetanHerve) [reply Modules/invoices/View/Invoiceglobal/editAction.php:110]: oh, of course

[the remark this answers is not in the document]
Comment (osallou) [reply Modules/invoices/View/Invoiceglobal/editAction.php:110]: ok, item is a pointer to data elt, so works

**Pull request merged**